frontend/test/account-send.test.ts

```diff
@@ -0,0 +1,309 @@
+const viewingPrivateKey = '0x290a15e2b46811c84a0c26624fd7fdc12e38143ae75518fc48375d41035ec5c1'; // this viewing key is taken from the testkeys in the umbra-js tests
+
+describe('encryptAccountData', () => {
+  invalidAddresses.forEach((recipientAddress) => {
+    it('Throws when given an invalid recipient address', () => {
+      const { pubKey, advancedMode, usePublicKeyChecked, viewingPrivateKey, encryptionCount } = randomData();
+      const x = () =>
+        encryptAccountData(
+          { recipientAddress, advancedMode, usePublicKeyChecked, pubKey },
+          { encryptionCount, viewingPrivateKey }
+        );
+      expect(x).toThrow('Invalid recipient address');
+    });
+  });
+
+  invalidEncryptionCounts.forEach((encryptionCount) => {
+    it('Throws when given an invalid encryption count', () => {
+      // Generate random data for everything except the encryption count.
+      const recipientAddress = ethers.Wallet.createRandom().address;
+      const pubKey = ethers.Wallet.createRandom().publicKey;
+      const advancedMode = Math.random() > 0.5;
+      const usePublicKeyChecked = Math.random() > 0.5;
+      const viewingPrivateKey = ethers.Wallet.createRandom().privateKey;
+      const x = () =>
+        encryptAccountData(
+          { recipientAddress, advancedMode, usePublicKeyChecked, pubKey },
+          { encryptionCount, viewingPrivateKey }
+        );
+      expect(x).toThrow('Invalid count provided for encryption');
+    });
+  });
+
+  invalidPrivateKeys.forEach((viewingPrivateKey) => {
+    const { recipientAddress, pubKey, advancedMode, usePublicKeyChecked, encryptionCount } = randomData();
+    it('Throws when given an invalid viewing key', () => {
+      const x = () =>
+        encryptAccountData(
+          { recipientAddress, advancedMode, pubKey, usePublicKeyChecked },
+          { encryptionCount, viewingPrivateKey }
+        );
+      expect(x).toThrow('Invalid viewing key');
+    });
+  });
+});
+
+describe('decryptData', () => {
+  invalidEncryptionCounts.forEach((encryptionCount) => {
+    it('Decryption invalid count', () => {
+      const { viewingPrivateKey, ciphertext } = randomData();
+      const x = () => decryptData(ciphertext, { encryptionCount, viewingPrivateKey });
+      expect(x).toThrow('Invalid count for decryption');
+    });
+  });
+
+  invalidPrivateKeys.forEach((viewingPrivateKey) => {
+    it('Decryption invalid viewing key', () => {
+      const { ciphertext, encryptionCount } = randomData();
+      const x = () => decryptData(ciphertext, { encryptionCount, viewingPrivateKey });
+      expect(x).toThrow('Invalid viewing key');
+    });
+  });
+
+  [
+    '',
+    '0xed72d644be0208e4d1c6312a04d2d623b99b2487f58d80f6169f9522d984cdb', // Too short.
+    '0xed72d644be0208e4d1c6312a04d2d623b99b2487f58d80f6169f9522d984cdbbb', // Too long.
+    'ffed72d644be0208e4d1c6312a04d2d623b99b2487f58d80f6169f9522d984cdbb', // Right length, wrong format.
```

Comment:
Maybe add a testcase with an invalid hex value? Like `xed72d644be0208e4d1c6312a04d2d623b99b2487f58d80f6169f9522d984cdbbz`

Reply:
Done in 70417bc